Add 2026-04-09 spell document (#88)
sky-ecosystem/executive-votes: 2026/executive-vote-2026-04-09-launch-avalanche-skylink-staking-rewards-update.md

---

> title: Template - [Executive Vote] Launch Avalanche SkyLink, Update Staking Rewards, Grove Genesis Capital Transfer, Safe Harbor Update, Prime Proxy Spells - April 9, 2026

Reviewer: ```suggestion
title: Template - [Executive Vote] Launch Avalanche SkyLink, Update Staking Rewards, Grove Genesis Capital Transfer, Safe Harbor Update, Prime Agent Proxy Spells - April 9, 2026
```

> summary: Launch SkyLink to Avalanche, update LSSKY-SKY staking rewards, transfer Genesis Capital to Grove, update the Safe Harbor agreement to account for the new SkyLink deployment, whitelist proxy spells for Spark and Grove.

Reviewer: Consistent with previous spells

```suggestion
summary: Launch SkyLink to Avalanche, update LSSKY->SKY staking rewards, transfer Genesis Capital to Grove, update the Safe Harbor agreement to account for the new SkyLink deployment, whitelist proxy spells for Spark and Grove.
```

date: 2026-04-09T00:00:00.000Z
address: "$spell_address"
---

# [Executive Proposal] Launch Avalanche SkyLink, Update Staking Rewards, Grove Genesis Capital Transfer, Safe Harbor Update, Prime Proxy Spells - April 9, 2026

The Core Facilitators, Dewiz, and Sidestream have placed an executive proposal into the voting system. SKY holders should vote for this proposal if they support the following alterations to the Sky Protocol.

If you are new to voting in the Sky Protocol, please see the [voting guide](https://manual.makerdao.com/governance/voting-in-makerdao/on-chain-governance) to learn how voting works.

---

## Executive Summary

If this executive proposal passes, the following **actions** will occur within the Sky Protocol:

- Avalanche SkyLink will be launched.

> - LSSKY-SKY staking rewards will be updated.

Reviewer: ```suggestion
- LSSKY->SKY staking rewards will be updated.
```

> - $TBD USDS will be transferred to Grove as Genesis Capital.

Reviewer: ```suggestion
- 20,797,477 USDS will be transferred to Grove as Genesis Capital.
```

Reviewer 2: Nit: should this be "as part of" their Genesis Capital to make it clear that the figure is still 25M?

- The chain list and contract list in the Safe Harbor Agreement will be updated.

> - Prime proxy spells for Spark and Grove will be whitelisted in their respective StarGuard modules.

Reviewer: ```suggestion
- Prime Agent proxy spells for Spark and Grove will be whitelisted in their respective StarGuard modules.
```

**Voting for this executive proposal will place your SKY in support of the actions outlined above.**

Unless otherwise noted, the actions listed above are subject to the [GSM Pause Delay](https://sky-atlas.io/#A.1.9.3.1). This means that if this executive proposal passes, the changes and additions listed above will only become active in the Sky Protocol after the GSM Pause Delay has expired. The GSM Pause Delay is currently set to [**24 hours**](https://sky-atlas.io/#A.1.9.3.1.2).

This executive proposal includes an office-hours modifier that means that it **can only be executed between 14:00 and 21:00 UTC, Monday - Friday**.

If this executive proposal does not pass within 30 days, then it will expire and can no longer have any effect on the Sky Protocol.

---

## Proposal Details

### Launch Avalanche SkyLink

- **Authorization**: [Governance Poll 1627](https://vote.sky.money/polling/QmW61Lnd)
- **Proposal**: [Atlas Edit Weekly Cycle Proposal](https://forum.skyeco.com/t/atlas-edit-weekly-cycle-proposal-week-of-2026-04-06/27822), [Technical Scope](https://forum.skyeco.com/t/skylink-bridge-to-avalanche/27825)

> If this executive proposal passes, then the Avalanche SkyLink will be launched through the following series of actions:

Reviewer 2: ```suggestion
If this executive proposal passes, then the Avalanche SkyLink will be launched through the following series of actions.
```

Convention is to use a colon when under the same header, period when the items have their own sections

- `address _oapp` being [LZ_GOV_SENDER](https://etherscan.io/address/0x27FC1DD771817b53bE48Dc28789533BEa53C9CCA) from chainlog
- `address _lib` being [0xbB2Ea70C9E858123480642Cf96acbcCE1372dCe1](https://etherscan.io/address/0xbB2Ea70C9E858123480642Cf96acbcCE1372dCe1) - see [LayerZero docs](https://docs.layerzero.network/v2/deployments/chains/ethereum)
- `SetConfigParam[] _params` being an array with two items:
  - First item: Executor parameters
    - `uint32 eid` being **30106** - see [LayerZero docs](https://docs.layerzero.network/v2/deployments/chains/avalanche)
    - `uint32 configType` being **1** - Executor type from [the source](https://etherscan.io/address/0xbB2Ea70C9E858123480642Cf96acbcCE1372dCe1#code#F1#L16)
    - `bytes config` being encoded [ExecutorConfig](https://github.com/LayerZero-Labs/LayerZero-v2/blob/9c741e7f9790639537b1710a203bcdfd73b0b9ac/packages/layerzero-v2/evm/messagelib/contracts/SendLibBase.sol#L23-L26) with:
      - `maxMessageSize` being **10_000**

>       - `executor` being [0x173272739Bd7Aa6e4e214714048a9fE699453059](etherscan.io/address/0x173272739Bd7Aa6e4e214714048a9fE699453059) - see [LayerZero docs](https://docs.layerzero.network/v2/deployments/chains/ethereum)

Reviewer: ```suggestion
      - `executor` being [0x173272739Bd7Aa6e4e214714048a9fE699453059](https://etherscan.io/address/0x173272739Bd7Aa6e4e214714048a9fE699453059) - see [LayerZero docs](https://docs.layerzero.network/v2/deployments/chains/ethereum)
```

  - Second item: ULN parameters
    - `uint32 eid` being **30106** - see [LayerZero docs](https://docs.layerzero.network/v2/deployments/chains/avalanche)
    - `uint32 configType` being **2** - ULN type from [the source](https://etherscan.io/address/0xbB2Ea70C9E858123480642Cf96acbcCE1372dCe1#code#F1#L17)
    - `bytes config` being encoded [UlnConfig](https://github.com/LayerZero-Labs/LayerZero-v2/blob/9c741e7f9790639537b1710a203bcdfd73b0b9ac/packages/layerzero-v2/evm/messagelib/contracts/uln/UlnBase.sol#L8-L16) with:
      - `uint64 confirmations` being **15** - default configuration, see [LayerZero docs](https://layerzeroscan.com/tools/defaults?version=V2&srcChainKey[0]=ethereum&dstChainKey[0]=avalanche)
      - `uint8 requiredDVNCount` being **255** (meaning NONE, [source](https://github.com/LayerZero-Labs/LayerZero-v2/blob/9c741e7f9790639537b1710a203bcdfd73b0b9ac/packages/layerzero-v2/evm/messagelib/contracts/uln/UlnBase.sol#L11))
      - `uint8 optionalDVNCount` being **7**
      - `uint8 optionalDVNThreshold` being **4**

>       - `address[] requiredDVNs` being an array with 0 addresses
>       - `address[] optionalDVNs` being an array with 7 addresses - see [LayerZero docs](https://docs.layerzero.network/v2/deployments/chains/ethereum):
>         - [0x589dEDbD617e0CBcB916A9223F4d1300c294236b](https://etherscan.io/address/0x589dEDbD617e0CBcB916A9223F4d1300c294236b) (LayerZero Labs)
>         - [0xa59BA433ac34D2927232918Ef5B2eaAfcF130BA5](https://etherscan.io/address/0xa59BA433ac34D2927232918Ef5B2eaAfcF130BA5) (Nethermind)
>         - [0xa4fE5A5B9A846458a70Cd0748228aED3bF65c2cd](https://etherscan.io/address/0xa4fE5A5B9A846458a70Cd0748228aED3bF65c2cd) (Canary),
>         - [0x373a6E5c0C4E89E24819f00AA37ea370917AAfF4](https://etherscan.io/address/0x373a6E5c0C4E89E24819f00AA37ea370917AAfF4) (Deutsche Telekom),
>         - [0x06559EE34D85a88317Bf0bfE307444116c631b67](https://etherscan.io/address/0x06559EE34D85a88317Bf0bfE307444116c631b67) (P2P),
>         - [0x380275805876Ff19055EA900CDb2B46a94ecF20D](https://etherscan.io/address/0x380275805876Ff19055EA900CDb2B46a94ecF20D) (Horizen),
>         - [0x58249a2Ec05c1978bF21DF1f5eC1847e42455CF4](https://etherscan.io/address/0x58249a2Ec05c1978bF21DF1f5eC1847e42455CF4) (Luganodes)
> 

Reviewer: ```suggestion
      - `address[] requiredDVNs` being an array with 0 addresses
      - `address[] optionalDVNs` being an array with 7 addresses - see [LayerZero docs](https://docs.layerzero.network/v2/deployments/chains/ethereum):
        - [0x589dEDbD617e0CBcB916A9223F4d1300c294236b](https://etherscan.io/address/0x589dEDbD617e0CBcB916A9223F4d1300c294236b) (LayerZero Labs),
        - [0xa59BA433ac34D2927232918Ef5B2eaAfcF130BA5](https://etherscan.io/address/0xa59BA433ac34D2927232918Ef5B2eaAfcF130BA5) (Nethermind),
        - [0xa4fE5A5B9A846458a70Cd0748228aED3bF65c2cd](https://etherscan.io/address/0xa4fE5A5B9A846458a70Cd0748228aED3bF65c2cd) (Canary),
        - [0x373a6E5c0C4E89E24819f00AA37ea370917AAfF4](https://etherscan.io/address/0x373a6E5c0C4E89E24819f00AA37ea370917AAfF4) (Deutsche Telekom),
        - [0x06559EE34D85a88317Bf0bfE307444116c631b67](https://etherscan.io/address/0x06559EE34D85a88317Bf0bfE307444116c631b67) (P2P),
        - [0x380275805876Ff19055EA900CDb2B46a94ecF20D](https://etherscan.io/address/0x380275805876Ff19055EA900CDb2B46a94ecF20D) (Horizen),
        - [0x58249a2Ec05c1978bF21DF1f5eC1847e42455CF4](https://etherscan.io/address/0x58249a2Ec05c1978bF21DF1f5eC1847e42455CF4) (Luganodes)

```

#### Wire `SUSDS_OFT` on Ethereum Mainnet with Avalanche Mainnet

Set `pauser` by calling `SUSDS_OFT`.[`setPauser`](https://github.com/sky-ecosystem/sky-oapp-oft/blob/5ad5cb6bbe624e2b1cb99acfe3e4140fa1c233b9/contracts/SkyOFTCore.sol#L156) with:

- `address _pauser` being [0x38d1114b4cE3e079CC0f627df6aC2776B5887776](https://etherscan.io/address/0x38d1114b4cE3e079CC0f627df6aC2776B5887776) (”LayerZero Freezer Multisig Address” from the [Atlas edit](https://forum.skyeco.com/t/atlas-edit-weekly-cycle-proposal-week-of-2025-11-03/27381))
- `bool _canPause` being **true** (required to enable the pauser)

Set `SkyOFTAdapterMintBurn(sUSDS)` as a peer on Avalanche by calling `SUSDS_OFT`[`.setPeer`](https://github.com/LayerZero-Labs/LayerZero-v2/blob/9c741e7f9790639537b1710a203bcdfd73b0b9ac/packages/layerzero-v2/evm/oapp/contracts/oapp/OAppCore.sol#L43) with:

- [`SUSDS_OFT`](https://etherscan.io/address/0x85A3FE4DA2a6cB98A5bdF62458B0dB8471B9f0f1) being the address from chainlog
- `uint32 _eid` being **30106** - see [LayerZero docs](https://docs.layerzero.network/v2/deployments/chains/avalanche)

> - `bytes32 _peer` being the new `SkyOFTAdapterMintBurn(sUSDS)` on Avalanche at [0x85A3FE4DA2a6cB98A5bdF62458B0dB8471B9f0f1](https://etherscan.io/address/0x85A3FE4DA2a6cB98A5bdF62458B0dB8471B9f0f1) padded with zeros

Reviewer: This address is wrong

```suggestion
- `bytes32 _peer` being the new `SkyOFTAdapterMintBurn(sUSDS)` on Avalanche at [0x7297D4811f088FC26bC5475681405B99b41E1FF9](https://snowscan.xyz/address/0x7297D4811f088FC26bC5475681405B99b41E1FF9) padded with zeros
```

    - `uint16 msgType` being **2** (Meaning SEND_AND_CALL, [source](https://etherscan.io/address/0x1e1d42781fc170ef9da004fb735f56f0276d01b8#readContract#F3))
    - `bytes options` being [encoded](https://github.com/LayerZero-Labs/LayerZero-v2/blob/9c741e7f9790639537b1710a203bcdfd73b0b9ac/packages/layerzero-v2/evm/oapp/contracts/oapp/libs/OptionsBuilder.sol#L55-L56):
      - `uint128 _gas` being **130_000** ([source](https://github.com/sky-ecosystem/sky-oapp-oft/blob/0baba10c77a5cdfcef8fe9611e38d4306ab827a7/oft.layerzero.mainnet.config.ts#L20))
      - `uint128 _value` being **0** ([source](https://github.com/sky-ecosystem/sky-oapp-oft/blob/0baba10c77a5cdfcef8fe9611e38d4306ab827a7/oft.layerzero.mainnet.config.ts#L21))

### LSSKY->SKY Staking Rewards Update

- **Authorization**: [Governance Poll 1627](https://vote.sky.money/polling/QmW61Lnd)
- **Proposal**: [Atlas Edit Weekly Cycle Proposal](https://forum.skyeco.com/t/atlas-edit-weekly-cycle-proposal-week-of-2026-04-06/27822), [BA Labs Calculation](https://forum.skyeco.com/t/lssky-to-sky-rewards-sky-rewards-for-sky-stakers-normalization-configuration/27721/6)

> If this executive proposal passes, then then the distribution of SKY token rewards to SKY stakers will be updated through the following actions:

Reviewer: ```suggestion
If this executive proposal passes, then the distribution of SKY token rewards to SKY stakers will be updated through the following actions:
```

- Call [TreasuryFundedFarmingInit.updateFarmVest()](https://github.com/sky-ecosystem/endgame-toolkit/blob/master/script/dependencies/treasury-funded-farms/TreasuryFundedFarmingInit.sol#L128) with the following parameters:
  - dist: [0x675671A8756dDb69F7254AFB030865388Ef699Ee](https://etherscan.io/address/0x675671A8756dDb69F7254AFB030865388Ef699Ee)
  - vestTot: **192,110,322 SKY**
  - vestBgn: **`block.timestamp`**
  - vestTau: **90 days**

### Grove Genesis Capital Transfer

- **Authorization**: [Atlas A.2.8.2.2.2.5.2](https://sky-atlas.io/#062fdb39-464e-4a5b-a44f-3462d2d38be5)
- **Proposal**: [Forum post](https://forum.skyeco.com/t/grove-genesis-capital-transfer/27828)

If this executive proposal passes, then **20,797,477 USDS** will be transferred to the [Grove SubProxy](https://etherscan.io/address/0x1369f7b2b38c76B6478c0f0E66D94923421891Ba) as part of their Genesis Capital allocation.

### Safe Harbor Update

- **Authorization**: [Atlas A.2.11.1.2.3](https://sky-atlas.io/#fcd868db-4a91-4ee0-baf5-1ebd40fc651e)

> If this executive proposal passes, the Safe Harbor agreement will be updated to include the Avalanche chain along with the new GovernanceOAppReceiver on Avalanche, L2GovernanceRelay on Avalanche, USDS proxy and implementation on Avalanche, sUSDS proxy and implementation on Avalanche, SkyOFTAdapterMintBurn(USDS) on Avalanche, SkyOFTAdapter(sUSDS) on mainnet, sUSDS proxy and implementation on Avalanche, and SkyOFTAdapterMintBurn(sUSDS) on Avalanche added by this spell.

Reviewer: duplicated

```suggestion
If this executive proposal passes, the Safe Harbor agreement will be updated to include the Avalanche chain along with the new GovernanceOAppReceiver on Avalanche, L2GovernanceRelay on Avalanche, USDS proxy and implementation on Avalanche, sUSDS proxy and implementation on Avalanche, SkyOFTAdapterMintBurn(USDS) on Avalanche, SkyOFTAdapter(sUSDS) on mainnet, and SkyOFTAdapterMintBurn(sUSDS) on Avalanche added by this spell.
```

> ### Prime Proxy Spells

Reviewer: ```suggestion
### Prime Agent Proxy Spells
```

    - maxAmount: **50 million**
    - slope: **250 million per day**

##### [Arbitrum] Spark Liquidity Layer - Update Rate Limits

- **Authorization**: [Snapshot Poll](https://snapshot.box/#/s:sparkfi.eth/proposal/0x6c889e7be8fba52d9cac4bd2e89c9bcb4ee1952afb40b555e87bf09062cb837f)
- **Proposal**: [Forum Proposal](https://forum.skyeco.com/t/april-9-2026-proposed-changes-to-spark-for-upcoming-spell/27804)

Various [rate limits](https://sky-atlas.io/#A.2.2.9.1.1.1.2.2) for the Spark Liquidity Layer on Arbitrum will be updated as follows:

- Aave Arbitrum USDC
  - Deposits
    - maxAmount: **0**
  - Withdrawals
    - maxAmount: **0**
- Fluid Arbitrum sUSDS
  - Deposits
    - maxAmount: **0**
  - Withdrawals
    - maxAmount: **0**

##### [Base] Spark Liquidity Layer - Update Rate Limits

- **Authorization**: [Snapshot Poll](https://snapshot.box/#/s:sparkfi.eth/proposal/0x6c889e7be8fba52d9cac4bd2e89c9bcb4ee1952afb40b555e87bf09062cb837f)
- **Proposal**: [Forum Proposal](https://forum.skyeco.com/t/april-9-2026-proposed-changes-to-spark-for-upcoming-spell/27804)

> Various [rate limits](https://sky-atlas.io/#A.2.2.9.1.1.1.2.2) for the Spark Liquidity Layer on Base will be updated as follows:
> 
> - Aave Base USDC
>   - Deposits
>     - maxAmount: 0
>   - Withdrawals
>     - maxAmount: 0
> - Fluid Base sUSDS
>   - Deposits
>     - maxAmount: 0
>   - Withdrawals
>     - maxAmount: 0
> 
> ##### [Ethereum] Spark Savings - Raise Deposit Caps for spUSDC, spUSDT, and spETH

Reviewer: update formatting for consistency

```suggestion
Various [rate limits](https://sky-atlas.io/#A.2.2.9.1.1.1.2.2) for the Spark Liquidity Layer on Base will be updated as follows:

- Aave Base USDC
  - Deposits
    - maxAmount: **0**
  - Withdrawals
    - maxAmount: **0**
- Fluid Base sUSDS
  - Deposits
    - maxAmount: **0**
  - Withdrawals
    - maxAmount: **0**

##### [Ethereum] Spark Savings - Raise Deposit Caps for spUSDC, spUSDT, and spETH
```

- **Authorization**: [Snapshot Poll](https://snapshot.box/#/s:sparkfi.eth/proposal/0x51bf5882e51b16c35ea596cf0ca69d52aeec912c1a72b84b6512d2d5f07a0167)
- **Proposal**: [Forum Proposal](https://forum.skyeco.com/t/april-9-2026-proposed-changes-to-spark-for-upcoming-spell/27804)

[Deposit caps](https://sky-atlas.io/#77918273-af7f-4c7e-80af-3f3d9d2097ec) for spUSDC, spUSDT, and spETH will be increased to the following limits:

- spUSDT
  - Deposit cap: **4 billion USDT**
- spUSDC
  - Deposit cap: **2 billion USDC**
- spETH
  - Deposit cap: **500,000 ETH**

##### [Ethereum] SparkLend - Claim SparkLend Reserves

- **Authorization**: [Atlas A.6.1.1.1.2.6.1.2.1.2.3 - Token Claim Authorization](https://sky-atlas.io/#A.6.1.1.1.2.6.1.2.1.2.3)
- **Proposal**: [Forum Post](https://forum.skyeco.com/t/april-9-2026-proposed-changes-to-spark-for-upcoming-spell/27804)

SparkLend reserves will be claimed as follows:

- Claim all reserves.
- Transfer USD stablecoin reserves to ALM Proxy.

> - Transfer non-USD stablecoin reserves to Spark Operations Multisig [0x2E1b01adABB8D4981863394bEa23a1263CBaeDfC](https://etherscan.io/address/0x2E1b01adABB8D4981863394bEa23a1263CBaeDfC) to be liquidated.

Reviewer 2: ```suggestion
- Claim all reserves
- Transfer USD stablecoin reserves to ALM Proxy
- Transfer non-USD stablecoin reserves to Spark Operations Multisig [0x2E1b01adABB8D4981863394bEa23a1263CBaeDfC](https://etherscan.io/address/0x2E1b01adABB8D4981863394bEa23a1263CBaeDfC) to be liquidated
```

Usually we cap sentences with periods, given that the technical scope items do not have this, it is easier to just remove these periods and keep it consistaint within this spell doc.

- syrupUSDC: [0x80ac24aA929eaF5013f6436cdA2a7ba190f5Cc0b](https://etherscan.io/address/0x80ac24aA929eaF5013f6436cdA2a7ba190f5Cc0b)
  - Underlying Asset: **USDC**
  - Deposits:
    - Max amount: **50 million USDC**
    - Slope: **50 million USDC per day**
  - Withdrawals:
    - Amount: **0 USDC**
  - Max Exchange Rate:
    - `setMaxExchangeRate(MAPLE_SYRUP_USDC, 1e6, 3e6)`

##### [Ethereum] Increase JTRSY Deposit Rate Limits

- **Authorization**: [Governance Poll 1627](https://vote.sky.money/polling/QmW61Lnd)
- **Proposal**: [Atlas Edit Weekly Cycle Proposal](https://forum.skyeco.com/t/atlas-edit-weekly-cycle-proposal-week-of-2026-04-06/27822), [Technical Scope](https://forum.skyeco.com/t/april-9th-2026-proposed-changes-to-grove-for-upcoming-spell/27801)

JTRSY deposit [rate limit](https://sky-atlas.io/#A.2.2.9.1.1.1.2.2) parameters will be updated as follows:

- JTRSY (V3): [0xFE6920eB6C421f1179cA8c8d4170530CDBdfd77A](https://etherscan.io/address/0xFE6920eB6C421f1179cA8c8d4170530CDBdfd77A)
  - Underlying Asset: USDC
  - Deposits:
    - Max amount: **500 million USDC**
    - Slope: **500 million USDC per day**
  - Withdrawals:

>     - Amount: **Unlimited**

Reviewer 2: ```suggestion
    - Amount: **unlimited**
```
